### ISSUE TITLE
Hurley gbParams p and q values for He giants

### ISSUE THREAD

**Issue by reinhold-willcox** [TeamCOMPAS/COMPAS#744]
**Describe the bug**
In Hurley+ 00, section 6.1, the discussion on He giants states that the Luminosity is a piecewise function of two power laws (similar to equation 37 for giant stars). 

L = min(B* Mcore^q, D* Mcore^p)  (eq. 37)
L = min(B* Mcore^3, D* Mcore^5)  (eq. 84)

Below (37), p and q are defined as piecewise functions of the total mass, and these p and q values appear in the later equations calculating the GB core mass and timescales (39-42). We use analogs of these later equations for the He giants, with various values swapped out. Namely, A_H becomes A_He, t_BGB becomes t_HeMS, L_BGB becomes L_THe, and the B and D values defined just below equation 87 get swapped in for the values of the same letters in (39-42). However in COMPAS, we do not recalculate (p, q) = (3, 5) for He giants. We use them correctly in the luminosity calculation in (84), but we do not change them in gbParams for use in (39-42). 

If this is in fact a bug, it will affect core mass growth for He giants, and consequently the luminosity, radius, and transition time between HeHG and HeGB stars. 

This is easy enough to implement in the code, and I am happy to do so, but could someone who is more familiar with Hurley please check that this is the correct approach?

**Label the issue**
`urgency_low`      - This issue is not urgent
`severity_minor`    - This is a minor bug with minimal impact

**Versioning (please complete the following information):**
 - COMPAS v02.27.01

**Comment by jeffriley** [comment]
The code is as @reinhold-willcox describes.  I think the code is consistent with the text in Hurley+2000 surrounding eq (84) - Hurley justifies the use of p = 3 and q = 5 for eq (84) in text prior to eq (84) then, in the paragraph immediately following eq (84), explains the replacements to be made in eqs (39-42) in some detail: he explicitly states which variables should be replaced with their analogs, but does not mention p and q.  I would have thought that if Hurley wanted p and q to be changed in eqs (39-42) for He giants he would have explicitly stated so given how detailed he is in stating what variables should be replaced, but it is possible that he just assumed the text immediately prior to eq (84) is sufficient to indicate the replacement is required.  It would seem to make sense that p and q should be changed per eq (84), but somebody above my pay-grade might need to decide this one.  Of course, we could always ask Hurley...

**Comment by reinhold-willcox** [comment]
Asking Hurley might be very sensible. I thought the same about only changing the parameters that are explicitly mentioned. But we do reset the B, D values, even though they are not explicitly mentioned in the text. 

I think (39-42) are just the calculations for the integration constants for 34, when L has the form of piecewise exponentials, so it makes sense to use updated B, D, p and q. The p and q for giant stars, defined based on the total mass relative to the maximum He flash mass, doesn't really apply to He stars (I would guess).

**Comment by ilyamandel** [comment]
Is the PR still being worked on?

**Comment by reinhold-willcox** [comment]
@ilyamandel Not actively, but it's definitely an outstanding (minor) bug. I don't think we should close it.

**Comment by ilyamandel** [comment]
@jeffriley -- since we know that the Hurley text can't always be trusted, and you've looked at the Hurley code recently, can you check what the SSE code does for this one?

**Comment by jeffriley** [comment]
I believe @reinhold-willcox is right.

This code:

    *
    * Set the Helium star GB parameters
    *
          GB(8) = 8.0d-05
          GB(3) = 4.1d+04
          GB(4) = 5.5d+04/(1.d0+0.4d0*mass**4)
          GB(5) = 5.d0
          GB(6) = 3.d0
          GB(7) = (GB(3)/GB(4))**(1.d0/(GB(5)-GB(6)))

in Hurley's sse code `star.f` is executed when stars switch to stellar type 7 (HeMS).

I have confirmed empirically that Hurley's sse code sets `p = 5.0` and `q = 3.0` for stellar types 7 and 8 (HeMS and HeHG). Inspection of the code (see above) indicates this is also true for stellar type 9 (HeGB), but I was not able to generate a HeGB star to test...

I have changed COMPAS so that we do the same thing as the Hurley sse code - we now set GB parameter p = 5.0, and q = 3.0 for HeMS, HeHG, and HeGB stars.  I have tested this (small test only) for HeMS and HeHG stars - in BSE mode - but, as for sse, I am not able to generate a HeGB star to test that.

There is no difference to the COMPAS output for the HeMS phase (GB params p and q are not used on the HeMS phase I don't think), and I can't see a difference on the HeHG phase - see HRD plot below:

![HRD_COMPARISON](https://github.com/TeamCOMPAS/COMPAS/assets/26860470/53c83c53-d5bc-469e-b9d5-0088e429bd57)

The plot above shows the HR diagram for the primary star of a binary for COMPAS v02.46.03 overlayed with my branch that has this fix in it - and difference on the HeMS and HeHG phases are imperceptible (to me anyway).

I'll go ahead an open a PR with the changes - if someone can generate an HeGB star to test let me know...
